libs/datasets/data_source.py

```diff
@@ -51,6 +58,21 @@ def source_tag(cls) -> taglib.Source:
 
     @classmethod
     def _check_data(cls, data: pd.DataFrame):
```

Comment:
I think `_check_data` is a bit misleading - I don't think I would expect this to modify the data - maybe `_check_and_remove_unexpected_data` would be a better name?

Reply:
Done